src/python/WMComponent/WorkQueueManager/WorkQueueManagerCleaner.py

```diff
@@ -47,9 +45,11 @@ def algorithm(self, parameters):
             # this will clean up whatever left over from above clean up.
             # also if the wq replication has problem it won't delay the killing jobs in condor
             # and updating wmbs status
-            abortedAndForceCompleteRequests = self.abortedAndForceCompleteWorkflowCache.getData()
+            # state lists which shouldn't be populated in wmbs. (To prevent creating work before WQE status updated)
+            # added completed list due to the race condition (The extra cost of the query is ~1sec per request)
+            requestsToKill = self.reqmgr2Svc.getRequestByStatusFromMemoryCache(["aborted", "aborted-completed", "force-complete", "completed"])
```

Comment:
I guess this one will also kill logcollect/cleanup subscriptions for completed (force-completed) workflows, no?
I'm wondering whether it would be better to have a different approach for WorkQueueManager, just to make sure we'll only acquire work that's valid. Something like
1. fetch a list of acquired/running-open/running-closed workflows
2. acquire anything that's in that list
3. anything else that is in wmbs - except logcollect and cleanup - gets removed.